openassessment/xblock/openassessmentblock.py

```diff
@@ -1333,6 +1357,34 @@ def publish_assessment_event(self, event_name, assessment, **kwargs):
             event_data
         )
 
+    @XBlock.json_handler
+    def reset_submission(self, data, suffix=""):  # pylint: disable=unused-argument
+        """
+        Reset the student's submission.
+
+        Args:
+            data (dict): Unused parameter. Defaults to {}.
+            suffix (str, optional): Unused parameter. Defaults to ''.
+
+        Returns:
+            dict: A dictionary indication the status with keys 'success' (bool) and 'msg' (str)
+        """
+        StudentModule = import_student_module()  # pylint: disable=invalid-name
+        block_user = self.runtime.service(self, "user").get_current_user()
+        username = block_user.opt_attrs.get("edx-platform.username")
+
+        try:
+            user = get_user_by_username_or_email(username)
+            reset_student_attempts(self.course_id, user, self.location, user, True)  # pylint: disable=no-member
+        except get_user_model().DoesNotExist as error:
```

Comment:
Should we do the same with get_user_model as with the StudentModule?

Reply:
Yes, we could do that